### PR TITLE
Add boundary tests

### PR DESCRIPTION
**Pull request by ddreimane**
Adding boundary tests to project. For some of the test cases new files were created, for some we added new test cases.

### CODE REVIEW

**Review by nightm4re94 — CHANGES_REQUESTED** [review]
Thank you for the contribution!
I have proposed a few slight changes, mostly regarding decimal precision.
Please address my comments before I can merge.

File: tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        BigDecimal lowerBound = number.getLowerBound();
+        BigDecimal currentValue = new BigDecimal("-456.0");
+        BigDecimal currentValue2 = new BigDecimal("0");
+        BigDecimal currentValue3 = new BigDecimal("456.0");
+
+        number.setLowerBound(currentValue);
+        assertFalse(lowerBound.compareTo(currentValue) < 0);
+
+        number.setLowerBound(currentValue2);
+        assertTrue(lowerBound.compareTo(currentValue2) == 0);
```
[the remark this answers is not in the document]
Comment (jluech) [reply tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java:67]: actually, it doesn't even make sense to test against `lowerbound` as it's being retrieved before setting the lower bound (test was still green when commenting out the setter).
I changed it to instead make a direct `assertEquals(currentValueX, number.getLowerBound())`. This also simplifies understanding and avoids using `compareTo()`

File: tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        adapter.initialized();
+    }
+
+
+
+    @Test
+    public void testSetCurrentValue() {
+        NumberAdjuster number = new NumberAdjuster(0,0,150,300,0,400,1,4);
+
+        BigDecimal newValue = new BigDecimal("123456.0");
```
[the remark this answers is not in the document]
Comment (jluech) [reply tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java:38]: +1 ;)

File: tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nightm4re94) [reply tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java:67]: you're right, sorry I overlooked that at first.

File: tests/de/gurkenlabs/litiengine/RandomTests.java [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 import java.util.Arrays;
 import java.util.List;
 
 import org.junit.jupiter.api.Test;
 
+import static org.mockito.ArgumentMatchers.anyDouble;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import static org.junit.jupiter.api.Assertions.*;
```
[the remark this answers is not in the document]
Comment (jluech) [reply tests/de/gurkenlabs/litiengine/RandomTests.java:14]: must have slipped in when optimizing imports with IntelliJ. Separated into required assertions

[the remark this answers is not in the document]
Comment (jluech) [reply tests/de/gurkenlabs/litiengine/RandomTests.java:14]: also cleaned up some tests around that IntelliJ warned about (replacing `assertTrue` for comparison with`assertEquals`)

File: src/de/gurkenlabs/litiengine/Align.java [hunk 4]
```diff
@@ -90,11 +92,11 @@ public int getValue(int width) {
    */
   public double getLocation(final double width, final double objectWidth) {
     double value = this.getValue(width);
-    double location = value - objectWidth / 2.0;
+    double location = BigDecimal.valueOf(value).subtract(BigDecimal.valueOf(objectWidth).divide(BigDecimal.valueOf(2.0))).doubleValue();
```
[the remark this answers is not in the document]
Comment (jluech) [reply src/de/gurkenlabs/litiengine/Align.java:95]: huh, didn't know about the delta option, nice to know!
I guess it was just for convenience... I reverted it to use simple double division and added a delta of 0.001 to each test

File: tests/de/gurkenlabs/litiengine/AlignTests.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        alignObject = Align.get("right"); // portion = 1
+    }
+
+    @Test
+    public void getLocation_OnPoint() {
+        // act
+        double onPointLocation = alignObject.getLocation(1.0, 1.0);
+
+        // assert
+        assertEquals(0.0, onPointLocation); // clamp(0.5, 0, 0)
```
[the remark this answers is not in the document]
Comment (jluech) [reply tests/de/gurkenlabs/litiengine/AlignTests.java:21]: see comment above, added a delta of 0.001 to each test

**Review by nightm4re94 — APPROVED** [review]
Beautiful, thank you very much!

**Pull request merged**